src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -243,17 +254,27 @@ public ProtectedItemResponse GetProtectedItem()
         /// <returns>Recovery point detail as returned by the service</returns>
         public CmdletModel.RecoveryPointBase GetRecoveryPointDetails()
         {
-            AzureVmItem item = ProviderData[GetRecoveryPointParams.Item]
+            AzureVmItem item = ProviderData[RecoveryPointParams.Item]
                 as AzureVmItem;
 
-            string recoveryPointId = ProviderData[GetRecoveryPointParams.RecoveryPointId].ToString();
+            string recoveryPointId = ProviderData[RecoveryPointParams.RecoveryPointId].ToString();
 
             Dictionary<UriEnums, string> uriDict = HelperUtils.ParseUri(item.Id);
             string containerUri = HelperUtils.GetContainerUri(uriDict, item.Id);
             string protectedItemName = HelperUtils.GetProtectedItemUri(uriDict, item.Id);
 
             var rpResponse = ServiceClientAdapter.GetRecoveryPointDetails(containerUri, protectedItemName, recoveryPointId);
-            return RecoveryPointConversions.GetPSAzureRecoveryPoints(rpResponse, item);
+
+            var rp = RecoveryPointConversions.GetPSAzureRecoveryPoints(rpResponse, item);
+
+            string keyFileDownloadLocation = (string)ProviderData[RecoveryPointParams.KeyFileDownloadLocation];
+            if (!string.IsNullOrEmpty(keyFileDownloadLocation))
+            {
+                string absoluteFilePath = System.IO.Path.Combine(keyFileDownloadLocation, "key.blob");
+                System.IO.File.WriteAllBytes(absoluteFilePath, Convert.FromBase64String(((AzureVmRecoveryPoint)rp).KeyAndSecretDetails.KeyBackupData));
```

Comment:
Line breaks.  Why do you have the fully naemspace qualified typoe name here, rather than using the namespace